WordPress/WordPressTest/BlockEditorSettingsServiceTests.swift

```diff
@@ -20,11 +20,10 @@ class BlockEditorSettingsServiceTests: XCTestCase {
         contextManager = TestContextManager()
         context = contextManager.mainContext
         mockRemoteApi = MockWordPressComRestApi()
-        blog = ModelTestHelper.insertDotComBlog(context: context)
-        blog.dotComID = NSNumber(value: 1)
-        blog.account?.authToken = "auth"
-        blog.setValue("5.8", forOption: "software_version")
-
+        blog = BlogBuilder(context)
+            .with(wordPressVersion: "5.8")
+            .withAnAccount()
+            .build()
```

Comment:
This is the code that addresses

> The error was a result of the code going down an unexpected path because the software version wasn't being properly picked up by the blog anymore.

How does this code addresses the issue? Is there more knowledge within `BlogBuilder` than in the previous setup approach? What part of the change in the hotfix surfaced this issue? (cc @emilylaguna)

Reply:
BTW I'm asking this to:

- Learn about this overwhelmingly vast codebase 😅 
- Check whether there are other parts of the test suite that should be updated as a followup to avoid these issues in the future (e.g. is `BlogBuilder` the way to go for this? Then maybe it should be adopted across the board?)

Reply:
> How does this code addresses the issue? Is there more knowledge within BlogBuilder than in the previous setup approach? 

I'm still looking into this to try and answer that question and will let you know what I find. But it's a curious case to me too because conceptually the old version of this code is doing the same thing as the new version.